cookbook/core/flyte_basics/files.py

```diff
@@ -2,68 +2,85 @@
+which are backed by the `blob type <https://github.com/lyft/flyteidl/blob/cee566b2e6e109120f1bb34c980b1cfaf006a473/protos/flyteidl/core/types.proto#L47>`__.
 
-Note that the type message includes an optional ``format`` field which is a text-field used to denote the file extension.
+Let's assume our mission here is pretty simple. We want to consider a couple of links, download the pictures, rotate them, and return the rotated images.
 """
 
+# %%
+# Let's first import the libraries.
 import os
-import urllib.request
 
 import cv2
 import flytekit
 from flytekit import task, workflow
-from flytekit.types.file import FlyteFile
+from flytekit.types.file import JPEGImageFile
 
 # %%
-# Let's assume our mission here is pretty simple. We want to take each of these links, download the picture, rotate it
-# and return the file.
-default_images = [
-    "https://upload.wikimedia.org/wikipedia/commons/a/a8/Fractal_pyramid.jpg",
-    "https://upload.wikimedia.org/wikipedia/commons/thumb/a/ad/Julian_fractal.jpg/256px-Julian_fractal.jpg",
-]
+# ``JPEGImageFile`` is a pre-formatted FlyteFile type. It is equivalent to ``FlyteFile[typing.TypeVar("jpeg")]``.
+#
+# .. note::
+#   ``FlyteFile`` literal can be scoped with a string, which gets inserted into the format of the Blob type ("jpeg" is the string in ``FlyteFile[typing.TypeVar("jpeg")]``). The ``[]`` are entirely optional, and if you don't specify it, the format will just be an ``""``.
 
 
 # %%
-# Note the signature of the return type of this task is a ``FlyteFile``. Files do not have a native object in Python
-# so we had to write one ourselves. There does exist the ``os.PathLike`` protocol, but nothing implements it.
-#
-# When this task finishes, the flytekit engine will detect the ``FlyteFile`` instance being returned, find a location
-# in Flyte's object store (usually S3), upload the file to that location, and create a Blob literal pointing to it.
-#
-# Note that the ``FlyteFile`` literal is scoped with a string, which gets inserted into the format of the Blob type.
-# The ``[]`` are entirely optional, and if you don't specify it, the format will just be an ``""``.
+# Next, we write a task that accepts a ``JPEGImageFile`` as an input and returns the rotated image as an output, which again is the ``JPEGImageFile``.
+# Files do not have a native object in Python, so we had to write one ourselves.
+# There does exist the ``os.PathLike`` protocol, but nothing implements it.
 @task
-def rotate(image_location: str) -> FlyteFile:
+def rotate(image_location: JPEGImageFile) -> JPEGImageFile:
     """
     Download the given image, rotate it by 180 degrees
     """
     working_dir = flytekit.current_context().working_directory
-    local_image = os.path.join(working_dir, "incoming.jpg")
-    urllib.request.urlretrieve(image_location, local_image)
-    img = cv2.imread(local_image, 0)
+    with open(image_location, "rb"):
+        ...
+    img = cv2.imread(str(image_location), 0)
     if img is None:
```

Comment:
This is odd

Reply:
cc @wild-endeavor can we not have a download method that returns the path?
https://pythonexamples.org/python-opencv-read-image-cv2-imread/
This is very awkward to use

Reply:
@wild-endeavor: Can you please check this?

Reply:
A download method on which object?

Reply:
@wild-endeavor I believe Ketan is referring to the file -- download method on the file object.